api/client/create.go

```diff
@@ -99,6 +112,12 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 		tag = tags.DefaultTag
 	}
 
+	if pull {
```

Comment:
@stevvooe  I don't get why placing here would bypass trust.  Placing this here still use the `trustedRef`, `cli.pullImageCustomOut(config.Image, cli.err)` pull the latest image and the next steps will check against trust. The `if pull` block is just like we do when the image is not exist on local, see from https://github.com/docker/docker/blob/master/api/client/create.go#L121 to https://github.com/docker/docker/blob/master/api/client/create.go#L131
Maybe I'm wrong...